src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -124,7 +120,7 @@ private static bool TryUseSubstring(ReadOnlySpan<string> uniqueStrings, bool ign
                 // actually perform the comparison as case-sensitive even if case-insensitive
                 // was requested, as there's nothing that would compare equally to the substring
                 // other than the substring itself.
-                bool canSwitchIgnoreCaseToCaseSensitive = ignoreCase;
+                bool canSwitchIgnoreCaseToCaseSensitive = true;
```

Comment:
This is demonstrably `true` because we're inside the [`if (ignoreCase)` immediately above](https://github.com/dotnet/runtime/pull/87590/files#diff-8dfef3756fbf6ca0aab7668e6abf54ae6c416c60cb1b3bd86455204942f7dbefR117)